examples/wellgraph.cpp

```diff
@@ -0,0 +1,134 @@
+    const auto groupnames = schedule.groupNames();
+    const std::size_t last = schedule.getTimeMap().last();
+    // Group -> Group relations.
+    for (const auto& gn : groupnames) {
+        const auto& g = schedule.getGroup(gn, last);
+        const auto& children = g.groups();
+        if (!children.empty()) {
+            os << "    " << gn << " -> {";
+            for (const auto& child : children) {
+                os << ' ' << child;
+            }
+            os << " }\n";
+        }
+    }
+    // Group -> Well relations.
+    os << "    node [shape=box]\n";
+    for (const auto& gn : groupnames) {
+        const auto& g = schedule.getGroup(gn, last);
+        const auto& children = g.wells();
+        if (!children.empty()) {
+            os << "    " << gn << " -> {";
+            for (const auto& child : children) {
+                os << ' ' << child;
+            }
+            os << " }\n";
+        }
+    }
+    // Color wells by injector or producer.
+    for (const auto& w : schedule.getWellsatEnd()) {
+        os << "    " << w.name();
+        if (w.isProducer() && w.isInjector()) {
+            os << " [color=purple]\n";
+        } else if (w.isProducer()) {
+            os << " [color=red]\n";
+        } else {
+            os << " [color=blue]\n";
+        }
+    }
+    os << "}\n";
+    std::cout << "complete." << std::endl;
+}
+
+
+inline Opm::Schedule loadSchedule(const std::string& deck_file)
+{
+    Opm::ParseContext parseContext({{Opm::ParseContext::PARSE_RANDOM_SLASH, Opm::InputError::IGNORE},
+                                    {Opm::ParseContext::PARSE_MISSING_DIMS_KEYWORD, Opm::InputError::WARN},
+                                    {Opm::ParseContext::SUMMARY_UNKNOWN_WELL, Opm::InputError::WARN},
+                                    {Opm::ParseContext::SUMMARY_UNKNOWN_GROUP, Opm::InputError::WARN}});
+    Opm::ErrorGuard errors;
+    Opm::Parser parser;
+    auto python = std::make_shared<Opm::Python>();
+
+    std::cout << "Loading and parsing deck: " << deck_file << " ..... "; std::cout.flush();
+    auto deck = parser.parseFile(deck_file, parseContext, errors);
+    std::cout << "complete.\n";
+
+    std::cout << "Creating EclipseState .... ";  std::cout.flush();
+    Opm::EclipseState state( deck );
+    std::cout << "complete.\n";
+
+    std::cout << "Creating Schedule .... ";  std::cout.flush();
+    Opm::Schedule schedule( deck, state, python);
+    std::cout << "complete." << std::endl;
+
+    return schedule;
+}
+
+
+int main(int argc, char** argv)
+{
+    std::ostringstream os;
+    std::shared_ptr<Opm::StreamLog> string_log = std::make_shared<Opm::StreamLog>(os, Opm::Log::DefaultMessageTypes);
+    Opm::OpmLog::addBackend( "STRING" , string_log);
+    try {
+        for (int iarg = 1; iarg < argc; iarg++) {
+            const std::string filename = argv[iarg];
+            const auto sched = loadSchedule(filename);
+            const auto casename = std::filesystem::path(filename).stem();
```

Comment:
I guess you should use the `Opm::filesystem` shim:
```C++
#include <opm/common/utility/FileSystem.hpp>
...
const auto casename = Opm::filesystem::path(filename).stem();
```

Reply:
Done